handler/producer/sequence.go

```diff
@@ -87,16 +121,40 @@ func pipe(req *http.Request, rt []Roundtrip, kind string) chan *Result {
 	return result
 }
 
-func pipeResult(ctx context.Context, req *http.Request, results chan *Result, rt Roundtrip) {
+func pipeResults(results, rch2 chan *Result) {
+	defer close(results)
+	for r := range rch2 {
+		results <- r
```

Comment:
select case with ctx.Done and return asap if true

Reply:
especially if called from `produceAndPipeResults` or `produceResults` if renamed.